### PR TITLE
feat(core): make smaller by removing lodash

### PR DESCRIPTION
**Pull request by Haroenv**
**Summary**

Lodash is no longer used upstream, meaning that it now got included for just two functions

**Result**

bundlesize is smaller (can't give a number, since the UMD marked lodash as a peerDependency)

### CODE REVIEW

File: src/highlight/highlight.ts [hunk 1]
```diff
@@ -1,7 +1,13 @@
-const get = require('lodash/get');
 import { Component, Input } from '@angular/core';
 import { bem } from '../utils';
 
+function getPropertyByPath(object: object, path: string): any {
```
Comment (francoischalifour) [inline src/highlight/highlight.ts:4]: Why not take the exact same [implementation](https://github.com/algolia/instantsearch.js/blob/9296022fecadfbf82f15e837c215a1356eac4bc5/src/lib/utils/getPropertyByPath.ts) used in IS.js and RIS?

**Review by francoischalifour — COMMENTED** [review]
These two new functions could go in [utils](https://github.com/algolia/angular-instantsearch/blob/65830ea77646bec4832d91e0890684daca3bed3f/src/utils.ts) so that we can reuse them.

File: src/highlight/highlight.ts (hunk 1 above)
Comment (Haroenv) [reply src/highlight/highlight.ts:4]: it's the implementation of React InstantSearch, which supports bracket-notation ([0])

Comment (eunjae-lee) [inline src/highlight/highlight.ts:4]: Where is this copied from? It'd good to put the link to the source.

**Review by eunjae-lee — APPROVED** [review]
Looks good to me.

File: src/utils.ts [hunk 2]
```diff
@@ -21,3 +21,36 @@ export function noop(...args: any[]): void {}
 export function capitalize(s) {
   return s.charAt(0).toUpperCase() + s.slice(1);
 }
+
+// source: instantsearch.js
```
Comment (francoischalifour) [inline src/utils.ts:25]:
```suggestion
// See https://github.com/algolia/instantsearch.js/blob/9296022fecadfbf82f15e837c215a1356eac4bc5/src/lib/utils/range.ts
```

File: src/utils.ts [hunk 3]
```diff
@@ -21,3 +21,36 @@ export function noop(...args: any[]): void {}
...
+  //   { start: 1, end: 5000, step: 500 }
+  //   => Array length = (5000 - 1) / 500 = 9.998
+  const arrayLength = Math.round((end - start) / limitStep);
+
+  return [...Array(arrayLength)].map(
+    (_, current) => start + current * limitStep
+  );
+}
+
+// source: react-instantsearch
```
Comment (francoischalifour) [inline src/utils.ts:50]:
```suggestion
// See https://github.com/algolia/react-instantsearch/blob/86dfe8674d566124af55a8f044051d0062786c1a/packages/react-instantsearch-core/src/core/utils.ts#L138-L142
```

**Review by francoischalifour — APPROVED** [review]

**Pull request merged**